### PR TITLE
[MPS] Add `tri[lu]_indices`

### PR DESCRIPTION
**Pull request by malfet**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137648
* #137647
* #137601

Requested in https://github.com/pytorch/pytorch/issues/77764#issuecomment-2402365980
Copy-n-paste kernel implementation from https://github.com/pytorch/pytorch/blob/13cf8360d87306ab3242e827eb662682cff17a19/aten/src/ATen/native/cuda/TensorFactories.cu#L92

though use `float` instead of `double` for square root computation

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/TriangularOps.mm [hunk 1]
```diff
@@ -1,20 +1,239 @@
 //  Copyright © 2022 Apple Inc.
```
Comment (Skylion007) [inline aten/src/ATen/native/mps/operations/TriangularOps.mm:1]: Hmm, legal nit but @malfet this copyright comment is not correct given the CLA, right?

Comment (malfet) [reply aten/src/ATen/native/mps/operations/TriangularOps.mm:1]: All copyright headers are superseded by top level license, unless it's a 3rd party code. But I don't want to fight this fight today...

**Review by Skylion007 — APPROVED** [review]

**Review by albanD — APPROVED** [review]
Sounds good!

**Pull request closed without merging**